### PR TITLE
Stop treating ECDH as experimental

### LINKED ISSUE #665: Review if ECDH is still experimental
ECDH is needed for Lightning, and it would be convenient to use the implementation here.

ECDH was [marked as experimental back in 2015](https://github.com/bitcoin-core/secp256k1/commit/83221ecb009dab11d9967abb9aa4aa8d27fdf477), and the API was last changed more than a year ago.

I am wondering how I can help make the ECDH implementation considered stable.

### PR DESCRIPTION
**Pull request by jonasnick**
Fixes #665

### CODE REVIEW

**Comment by sipa** [comment]
ACK, I think it's well tested and the API is sufficiently generic that I don't expect it needing changes any time soon.

**Comment by elichai** [comment]
Nice, ACK.
so after this only ARM assembly is experimental here?

**Comment by real-or-random** [comment]
ACK 5a81a2ac249fc7a06517a1bc970c537f99c3a330

**Comment by jonasnick** [comment]
rebased

**Comment by jonasnick** [comment]
> so after this only ARM assembly is experimental here?

and schnorrsig :)

File: .travis.yml [hunk 1]
```diff
@@ -36,8 +36,8 @@ env:
     - CFLAGS=-O0 CTIMETEST=
     - ECMULTGENPRECISION=2
     - ECMULTGENPRECISION=8
-    - VALGRIND=yes ENDOMORPHISM=yes BIGNUM=no ASM=x86_64 EXPERIMENTAL=yes ECDH=yes  RECOVERY=yes EXTRAFLAGS="--disable-openssl-tests" CPPFLAGS=-DVALGRIND BUILD=
-    - VALGRIND=yes                  BIGNUM=no ASM=x86_64 EXPERIMENTAL=yes ECDH=yes  RECOVERY=yes EXTRAFLAGS="--disable-openssl-tests" CPPFLAGS=-DVALGRIND BUILD=
+    - VALGRIND=yes ENDOMORPHISM=yes BIGNUM=no ASM=x86_64 ECDH=yes  RECOVERY=yes EXTRAFLAGS="--disable-openssl-tests" CPPFLAGS=-DVALGRIND BUILD=
+    - VALGRIND=yes                  BIGNUM=no ASM=x86_64 ECDH=yes  RECOVERY=yes EXTRAFLAGS="--disable-openssl-tests" CPPFLAGS=-DVALGRIND BUILD=
```
Comment (elichai) [inline .travis.yml:40]: Can you leave the experimental but add `SCHNORRSIG=yes` in both of these? (I can open a separate PR for that if you prefer)

File: .travis.yml [hunk 2]
```diff
@@ -85,7 +85,7 @@ matrix:
             - libc6-dbg:i386
     # S390x build (big endian system)
     - compiler: gcc
-      env: HOST=s390x-unknown-linux-gnu ECDH=yes RECOVERY=yes EXPERIMENTAL=yes CTIMETEST=
+      env: HOST=s390x-unknown-linux-gnu ECDH=yes RECOVERY=yes CTIMETEST=
```
Comment (elichai) [inline .travis.yml:88]: Can you leave the experimental but add `SCHNORRSIG=yes` in both of these? (I can open a separate PR for that if you prefer)

File: .travis.yml (hunk 1 above)
Comment (jonasnick) [reply .travis.yml]: good catch, done

File: .travis.yml (hunk 2 above)
Comment (jonasnick) [reply .travis.yml]: good catch, done

**Comment by elichai** [comment]
ACK 14392d64e60d4beca8e9d9c531d11b1e6a95e58e

**Review by real-or-random — APPROVED** [review]
ACK 14392d64e60d4beca8e9d9c531d11b1e6a95e58e code review, did not test

**Comment by sipa** [comment]
utACK 14392d64e60d4beca8e9d9c531d11b1e6a95e58e, but needs rebase

**Comment by jonasnick** [comment]
rebased

**Review by real-or-random — APPROVED** [review]
ACK e6e3d5da2f92f327019a62dc5f62f9335b2d0975

**Review by elichai — APPROVED** [review]
ACK 353dff156f1ed43a93c14a3686dc0649713d2997

**Pull request merged**